Remove ENABLE_ACCESSIBILITY directive

annevk opened the pull request:
#### e49f6085669f2cc10c53fd665871ee2a6886ddf2
<pre>
Remove ENABLE_ACCESSIBILITY directive
<a href="https://bugs.webkit.org/show_bug.cgi?id=266971">https://bugs.webkit.org/show_bug.cgi?id=266971</a>

Reviewed by Michael Catanzaro.

It is finally enabled everywhere.

* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::detachWrapper):
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AXObjectCache.h:
(WebCore::AXComputedObjectAttributeCache::getIgnored const): Deleted.
(WebCore::AccessibilityReplacedText::AccessibilityReplacedText): Deleted.
(WebCore::AccessibilityReplacedText::postTextStateChangeNotification): Deleted.
(WebCore::AXComputedObjectAttributeCache::setIgnored): Deleted.
(WebCore::AXObjectCache::AXObjectCache): Deleted.
(WebCore::AXObjectCache::~AXObjectCache): Deleted.
(WebCore::AXObjectCache::get): Deleted.
(WebCore::AXObjectCache::getOrCreate): Deleted.
(WebCore::AXObjectCache::create): Deleted.
(WebCore::AXObjectCache::rootObject): Deleted.
(WebCore::AXObjectCache::rootObjectForFrame): Deleted.
(WebCore::AXObjectCache::focusedObjectForPage): Deleted.
(WebCore::AXObjectCache::enableAccessibility): Deleted.
(WebCore::AXObjectCache::setForceDeferredSpellChecking): Deleted.
(WebCore::AXObjectCache::disableAccessibility): Deleted.
(WebCore::AXObjectCache::setEnhancedUserInterfaceAccessibility): Deleted.
(WebCore::nodeHasRole): Deleted.
(WebCore::nodeHasGridRole): Deleted.
(WebCore::nodeHasCellRole): Deleted.
(WebCore::AXObjectCache::startCachingComputedObjectAttributesUntilTreeMutates): Deleted.
(WebCore::AXObjectCache::stopCachingComputedObjectAttributes): Deleted.
(WebCore::isNodeAriaVisible): Deleted.
(WebCore::AXObjectCache::rootAXEditableElement): Deleted.
(WebCore::AXObjectCache::modalNode): Deleted.
(WebCore::AXObjectCache::attachWrapper): Deleted.
(WebCore::AXObjectCache::checkedStateChanged): Deleted.
(WebCore::AXObjectCache::autofillTypeChanged): Deleted.
(WebCore::AXObjectCache::childrenChanged): Deleted.
(WebCore::AXObjectCache::onSelectedChanged): Deleted.
(WebCore::AXObjectCache::onTextSecurityChanged): Deleted.
(WebCore::AXObjectCache::onTitleChange): Deleted.
(WebCore::AXObjectCache::onValidityChange): Deleted.
(WebCore::AXObjectCache::onTextCompositionChange): Deleted.
(WebCore::AXObjectCache::valueChanged): Deleted.
(WebCore::AXObjectCache::onFocusChange): Deleted.
(WebCore::AXObjectCache::onPageActivityStateChange): Deleted.
(WebCore::AXObjectCache::onPopoverToggle): Deleted.
(WebCore::AXObjectCache::onScrollbarFrameRectChange): Deleted.
(WebCore::AXObjectCache::onWidgetVisibilityChanged): Deleted.
(WebCore::AXObjectCache::deferRecomputeIsIgnoredIfNeeded): Deleted.
(WebCore::AXObjectCache::deferRecomputeIsIgnored): Deleted.
(WebCore::AXObjectCache::deferTextChangedIfNeeded): Deleted.
(WebCore::AXObjectCache::deferSelectedChildrenChangedIfNeeded): Deleted.
(WebCore::AXObjectCache::deferTextReplacementNotificationForTextControl): Deleted.
(WebCore::AXObjectCache::deferRecomputeTableCellSlots): Deleted.
(WebCore::AXObjectCache::deferRowspanChange): Deleted.
(WebCore::AXObjectCache::detachWrapper): Deleted.
(WebCore::AXObjectCache::focusCurrentModal): Deleted.
(WebCore::AXObjectCache::performCacheUpdateTimerFired): Deleted.
(WebCore::AXObjectCache::frameLoadingEventNotification): Deleted.
(WebCore::AXObjectCache::frameLoadingEventPlatformNotification): Deleted.
(WebCore::AXObjectCache::handleAriaExpandedChange): Deleted.
(WebCore::AXObjectCache::deferModalChange): Deleted.
(WebCore::AXObjectCache::handleAllDeferredChildrenChanged): Deleted.
(WebCore::AXObjectCache::handleRoleChanged): Deleted.
(WebCore::AXObjectCache::deferAttributeChangeIfNeeded): Deleted.
(WebCore::AXObjectCache::handleAttributeChange): Deleted.
(WebCore::AXObjectCache::shouldProcessAttributeChange): Deleted.
(WebCore::AXObjectCache::handleFocusedUIElementChanged): Deleted.
(WebCore::AXObjectCache::handleScrollbarUpdate): Deleted.
(WebCore::AXObjectCache::onScrollbarUpdate): Deleted.
(WebCore::AXObjectCache::handleScrolledToAnchor): Deleted.
(WebCore::AXObjectCache::liveRegionChangedNotificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::notificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::passwordNotificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::performDeferredCacheUpdate): Deleted.
(WebCore::AXObjectCache::postLiveRegionChangeNotification): Deleted.
(WebCore::AXObjectCache::postNotification): Deleted.
(WebCore::AXObjectCache::announce): Deleted.
(WebCore::AXObjectCache::postTextReplacementNotification): Deleted.
(WebCore::AXObjectCache::postTextReplacementNotificationForTextControl): Deleted.
(WebCore::AXObjectCache::postTextStateChangeNotification): Deleted.
(WebCore::AXObjectCache::recomputeIsIgnored): Deleted.
(WebCore::AXObjectCache::handleTextChanged): Deleted.
(WebCore::AXObjectCache::handleRecomputeCellSlots): Deleted.
(WebCore::AXObjectCache::onRendererCreated): Deleted.
(WebCore::AXObjectCache::updateLoadingProgress): Deleted.
(WebCore::AXObjectCache::rangeForNodeContents): Deleted.
(WebCore::AXObjectCache::relatedObjectIDsFor): Deleted.
(WebCore::AXObjectCache::updateRelations): Deleted.
(WebCore::AXObjectCache::remove): Deleted.
(WebCore::AXObjectCache::selectedChildrenChanged): Deleted.
(WebCore::AXObjectCache::setIsSynchronizingSelection): Deleted.
(WebCore::AXObjectCache::setTextSelectionIntent): Deleted.
(WebCore::AXObjectCache::rangeForUnorderedCharacterOffsets): Deleted.
(WebCore::AXObjectCache::absoluteCaretBoundsForCharacterOffset): Deleted.
(WebCore::AXObjectCache::characterOffsetForIndex): Deleted.
(WebCore::AXObjectCache::startOrEndCharacterOffsetForRange): Deleted.
(WebCore::AXObjectCache::endCharacterOffsetOfLine): Deleted.
(WebCore::AXObjectCache::nextCharacterOffset): Deleted.
(WebCore::AXObjectCache::previousCharacterOffset): Deleted.
(WebCore::AXObjectCache::textMarkerDataForVisiblePosition): Deleted.
(WebCore::AXObjectCache::textMarkerDataForCharacterOffset): Deleted.
(WebCore::AXObjectCache::visiblePositionForTextMarkerData): Deleted.
(WebCore::AXObjectCache::visiblePositionFromCharacterOffset): Deleted.
(WebCore::AXObjectCache::postTextStateChangePlatformNotification): Deleted.
(WebCore::AXObjectCache::postTextReplacementPlatformNotification): Deleted.
(WebCore::AXObjectCache::textChangeForEditType): Deleted.
(WebCore::AXObjectCache::nodeTextChangePlatformNotification): Deleted.
(WebCore::AXObjectCache::treeData): Deleted.
(WebCore::AXAttributeCacheEnabler::AXAttributeCacheEnabler): Deleted.
(WebCore::AXAttributeCacheEnabler::~AXAttributeCacheEnabler): Deleted.
* Source/WebCore/accessibility/AccessibilityMenuList.cpp:
(WebCore::AccessibilityMenuList::didUpdateActiveOption):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::isDetached const):
(WebCore::AccessibilityObject::lineForPosition const):
(WebCore::AccessibilityObject::updateBackingStore):
(WebCore::AccessibilityObject::children):
(WebCore::AccessibilityObject::localizedActionVerb const):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::previousLineStartPosition const):
(WebCore::AccessibilityObject::accessibilityIgnoreAttachment const): Deleted.
(WebCore::AccessibilityObject::accessibilityPlatformIncludesObject const): Deleted.
(WebCore::AccessibilityObject::children): Deleted.
(WebCore::AccessibilityObject::localizedActionVerb const): Deleted.
(WebCore::AccessibilityObject::actionVerb const): Deleted.
(WebCore::AccessibilityObject::lineForPosition const): Deleted.
(WebCore::AccessibilityObject::updateBackingStore): Deleted.
(WebCore::AccessibilityObject::detachPlatformWrapper): Deleted.
* Source/WebCore/accessibility/cocoa/AccessibilityObjectCocoa.mm:
* Source/WebCore/accessibility/ios/AXObjectCacheIOS.mm:
* Source/WebCore/accessibility/ios/AccessibilityObjectIOS.mm:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/AccessibilityObjectMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/accessibility/playstation/AXObjectCachePlayStation.cpp:
* Source/WebCore/accessibility/playstation/AccessibilityObjectPlayStation.cpp:
* Source/WebCore/accessibility/win/AccessibilityObjectWin.cpp:
* Source/WebCore/accessibility/win/AccessibilityObjectWrapperWin.cpp:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::updateTitle):
(WebCore::Document::willBeRemovedFromFrame):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/TextCheckingHelper.cpp:
(WebCore::platformOrClientDrivenTextCheckerEnabled):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::showPopover):
(WebCore::HTMLElement::hidePopoverInternal):
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::HTMLTextFormControlElement::setInnerTextValue):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewAccessible.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewAccessible.h:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseDispose):
(webkit_web_view_base_class_init):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::accessible):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::View::~View):
(WKWPE::View::accessible const):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindA11y):
(WebKit::bubblewrapSpawn):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::accessibilityProxy):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::bindAccessibilityTree):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKAccessibilityEnable):
(WKAccessibilityRootObject):
(WKAccessibilityFocusedObject):
(WKAccessibilityFocusedUIElement):
(WKAccessibilityAnnounce):
(WKAccessibilityEnableEnhancedAccessibility):
(WKAccessibilityEnhancedAccessibilityEnabled):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame setAccessibleName:]):
(-[WebFrame enhancedAccessibilityEnabled]):
(-[WebFrame setEnhancedAccessibility:]):
(-[WebFrame accessibilityRoot]):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/DumpRenderTree/AccessibilityController.cpp:
(AccessibilityController::resetToConsistentState):
* Tools/DumpRenderTree/AccessibilityUIElement.cpp:
(AccessibilityUIElement::getJSClass):
* Tools/MiniBrowser/wpe/main.cpp:
(activate):
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.h:
(WTR::AccessibilityController::rootElement): Deleted.
(WTR::AccessibilityController::focusedElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarker.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarkerRange.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityController.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityTextMarker.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityTextMarkerRange.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityUIElement.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::beginTesting):
(WTR::InjectedBundle::done):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
(WTR::InjectedBundle::textInputController):
(WTR::InjectedBundle::accessibilityController):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::didClearWindowForFrame):
* Tools/WebKitTestRunner/InjectedBundle/mac/AccessibilityUIElementMac.mm:
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityControllerWin.cpp:
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityUIElementWin.cpp:
* Tools/wpe/backends/PlatformWPE.cmake:
* Tools/wpe/backends/ViewBackend.cpp:
(WPEToolingBackends::ViewBackend::dispatchInputKeyboardEvent):
(WPEToolingBackends::ViewBackend::initializeAccessibility): Deleted.
(WPEToolingBackends::ViewBackend::updateAccessibilityState): Deleted.
* Tools/wpe/backends/ViewBackend.h:
* Tools/wpe/backends/atk/ViewBackendAtk.cpp:
* Tools/wpe/backends/atk/WebKitAccessibleApplication.cpp:
(webkitAccessibleApplicationSetChild):
* Tools/wpe/backends/atk/WebKitAccessibleApplication.h:

Canonical link: <a href="https://commits.webkit.org/273021@main">https://commits.webkit.org/273021@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f66f2737263c30ae24031944947259304770c314

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33980 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12768 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35941 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36609 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30791 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35044 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15152 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9910 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29843 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34474 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10794 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30311 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9412 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9515 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30330 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37917 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/28912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30861 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30652 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35627 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/33895 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9643 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7579 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33527 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11430 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/40444 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7827 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10212 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8450 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10454 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/OptionsWPE.cmake:
```diff
@@ -226,15 +225,13 @@ set(LIB_INSTALL_DIR "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE PATH "Absolute path to
+find_package(ATKBridge)
+if (NOT ATKBridge_FOUND)
+    message(FATAL_ERROR "at-spi2-atk is needed for accessibility")
 endif ()
```
Same here as before: move these find_package() to the top of the file, add `REQUIRED` to both of them, and remove the custom error messages as they're no longer needed. The only point of those custom messages is to give a hint that the feature can be disabled, which is no longer true.

mcatanzaro commented on Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
```diff
@@ -54,7 +54,7 @@ ADD_WK2_TEST(TestWebKitFaviconDatabase ${TOOLS_DIR}/TestWebKitAPI/Tests/WebKitGL
-if (ENABLE_ACCESSIBILITY AND ATSPI_FOUND)
+if (ATSPI_FOUND)
     ADD_WK2_TEST(TestWebKitAccessibility ${TOOLS_DIR}/TestWebKitAPI/Tests/WebKitGtk/TestWebKitAccessibility.cpp)
 endif ()
```
Don't need this conditional anymore: it's now always true

mcatanzaro reviewed (commented):
I'm not sure whether Igalia is OK with making WPE hard depend on ATK. It seems fine to me, but I will ask @carlosgcampos and @aperezdc about this. Advantage: easier for us to have fewer build flags. Disadvantage: ATK isn't used by GTK or WebKitGTK anymore, so it's maybe not the healthiest dependency for WPE.

(We should definitely remove `ENABLE(ACCESSIBILITY)` regardless. We can give WPE a `USE(ATK)` option instead if requiring ATK is not desired.)

mcatanzaro reviewed (approved):
In that case, this is good. Thank you!

mcatanzaro commented on Source/cmake/OptionsWPE.cmake:
```diff
@@ -11,6 +11,8 @@ endif ()
 set(USER_AGENT_BRANDING "" CACHE STRING "Branding to add to user agent string")
 
+find_package(ATK 2.16.0)
+find_package(ATKBridge)
```
```suggestion
find_package(ATK 2.16.0 REQUIRED)
find_package(ATKBridge REQUIRED)
```

mcatanzaro replied:
(We can remove them later when adding `USE(ATK)`, but for now, they are required.)

annevk committed:
Remove ENABLE_ACCESSIBILITY directive
https://bugs.webkit.org/show_bug.cgi?id=266971

Reviewed by Michael Catanzaro.

It is finally enabled everywhere.

* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::detachWrapper):
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AXObjectCache.h:
(WebCore::AXComputedObjectAttributeCache::getIgnored const): Deleted.
(WebCore::AccessibilityReplacedText::AccessibilityReplacedText): Deleted.
(WebCore::AccessibilityReplacedText::postTextStateChangeNotification): Deleted.
(WebCore::AXComputedObjectAttributeCache::setIgnored): Deleted.
(WebCore::AXObjectCache::AXObjectCache): Deleted.
(WebCore::AXObjectCache::~AXObjectCache): Deleted.
(WebCore::AXObjectCache::get): Deleted.
(WebCore::AXObjectCache::getOrCreate): Deleted.
(WebCore::AXObjectCache::create): Deleted.
(WebCore::AXObjectCache::rootObject): Deleted.
(WebCore::AXObjectCache::rootObjectForFrame): Deleted.
(WebCore::AXObjectCache::focusedObjectForPage): Deleted.
(WebCore::AXObjectCache::enableAccessibility): Deleted.
(WebCore::AXObjectCache::setForceDeferredSpellChecking): Deleted.
(WebCore::AXObjectCache::disableAccessibility): Deleted.
(WebCore::AXObjectCache::setEnhancedUserInterfaceAccessibility): Deleted.
(WebCore::nodeHasRole): Deleted.
(WebCore::nodeHasGridRole): Deleted.
(WebCore::nodeHasCellRole): Deleted.
(WebCore::AXObjectCache::startCachingComputedObjectAttributesUntilTreeMutates): Deleted.
(WebCore::AXObjectCache::stopCachingComputedObjectAttributes): Deleted.
(WebCore::isNodeAriaVisible): Deleted.
(WebCore::AXObjectCache::rootAXEditableElement): Deleted.
(WebCore::AXObjectCache::modalNode): Deleted.
(WebCore::AXObjectCache::attachWrapper): Deleted.
(WebCore::AXObjectCache::checkedStateChanged): Deleted.
(WebCore::AXObjectCache::autofillTypeChanged): Deleted.
(WebCore::AXObjectCache::childrenChanged): Deleted.
(WebCore::AXObjectCache::onSelectedChanged): Deleted.
(WebCore::AXObjectCache::onTextSecurityChanged): Deleted.
(WebCore::AXObjectCache::onTitleChange): Deleted.
(WebCore::AXObjectCache::onValidityChange): Deleted.
(WebCore::AXObjectCache::onTextCompositionChange): Deleted.
(WebCore::AXObjectCache::valueChanged): Deleted.
(WebCore::AXObjectCache::onFocusChange): Deleted.
(WebCore::AXObjectCache::onPageActivityStateChange): Deleted.
(WebCore::AXObjectCache::onPopoverToggle): Deleted.
(WebCore::AXObjectCache::onScrollbarFrameRectChange): Deleted.
(WebCore::AXObjectCache::onWidgetVisibilityChanged): Deleted.
(WebCore::AXObjectCache::deferRecomputeIsIgnoredIfNeeded): Deleted.
(WebCore::AXObjectCache::deferRecomputeIsIgnored): Deleted.
(WebCore::AXObjectCache::deferTextChangedIfNeeded): Deleted.
(WebCore::AXObjectCache::deferSelectedChildrenChangedIfNeeded): Deleted.
(WebCore::AXObjectCache::deferTextReplacementNotificationForTextControl): Deleted.
(WebCore::AXObjectCache::deferRecomputeTableCellSlots): Deleted.
(WebCore::AXObjectCache::deferRowspanChange): Deleted.
(WebCore::AXObjectCache::detachWrapper): Deleted.
(WebCore::AXObjectCache::focusCurrentModal): Deleted.
(WebCore::AXObjectCache::performCacheUpdateTimerFired): Deleted.
(WebCore::AXObjectCache::frameLoadingEventNotification): Deleted.
(WebCore::AXObjectCache::frameLoadingEventPlatformNotification): Deleted.
(WebCore::AXObjectCache::handleAriaExpandedChange): Deleted.
(WebCore::AXObjectCache::deferModalChange): Deleted.
(WebCore::AXObjectCache::handleAllDeferredChildrenChanged): Deleted.
(WebCore::AXObjectCache::handleRoleChanged): Deleted.
(WebCore::AXObjectCache::deferAttributeChangeIfNeeded): Deleted.
(WebCore::AXObjectCache::handleAttributeChange): Deleted.
(WebCore::AXObjectCache::shouldProcessAttributeChange): Deleted.
(WebCore::AXObjectCache::handleFocusedUIElementChanged): Deleted.
(WebCore::AXObjectCache::handleScrollbarUpdate): Deleted.
(WebCore::AXObjectCache::onScrollbarUpdate): Deleted.
(WebCore::AXObjectCache::handleScrolledToAnchor): Deleted.
(WebCore::AXObjectCache::liveRegionChangedNotificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::notificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::passwordNotificationPostTimerFired): Deleted.
(WebCore::AXObjectCache::performDeferredCacheUpdate): Deleted.
(WebCore::AXObjectCache::postLiveRegionChangeNotification): Deleted.
(WebCore::AXObjectCache::postNotification): Deleted.
(WebCore::AXObjectCache::announce): Deleted.
(WebCore::AXObjectCache::postTextReplacementNotification): Deleted.
(WebCore::AXObjectCache::postTextReplacementNotificationForTextControl): Deleted.
(WebCore::AXObjectCache::postTextStateChangeNotification): Deleted.
(WebCore::AXObjectCache::recomputeIsIgnored): Deleted.
(WebCore::AXObjectCache::handleTextChanged): Deleted.
(WebCore::AXObjectCache::handleRecomputeCellSlots): Deleted.
(WebCore::AXObjectCache::onRendererCreated): Deleted.
(WebCore::AXObjectCache::updateLoadingProgress): Deleted.
(WebCore::AXObjectCache::rangeForNodeContents): Deleted.
(WebCore::AXObjectCache::relatedObjectIDsFor): Deleted.
(WebCore::AXObjectCache::updateRelations): Deleted.
(WebCore::AXObjectCache::remove): Deleted.
(WebCore::AXObjectCache::selectedChildrenChanged): Deleted.
(WebCore::AXObjectCache::setIsSynchronizingSelection): Deleted.
(WebCore::AXObjectCache::setTextSelectionIntent): Deleted.
(WebCore::AXObjectCache::rangeForUnorderedCharacterOffsets): Deleted.
(WebCore::AXObjectCache::absoluteCaretBoundsForCharacterOffset): Deleted.
(WebCore::AXObjectCache::characterOffsetForIndex): Deleted.
(WebCore::AXObjectCache::startOrEndCharacterOffsetForRange): Deleted.
(WebCore::AXObjectCache::endCharacterOffsetOfLine): Deleted.
(WebCore::AXObjectCache::nextCharacterOffset): Deleted.
(WebCore::AXObjectCache::previousCharacterOffset): Deleted.
(WebCore::AXObjectCache::textMarkerDataForVisiblePosition): Deleted.
(WebCore::AXObjectCache::textMarkerDataForCharacterOffset): Deleted.
(WebCore::AXObjectCache::visiblePositionForTextMarkerData): Deleted.
(WebCore::AXObjectCache::visiblePositionFromCharacterOffset): Deleted.
(WebCore::AXObjectCache::postTextStateChangePlatformNotification): Deleted.
(WebCore::AXObjectCache::postTextReplacementPlatformNotification): Deleted.
(WebCore::AXObjectCache::textChangeForEditType): Deleted.
(WebCore::AXObjectCache::nodeTextChangePlatformNotification): Deleted.
(WebCore::AXObjectCache::treeData): Deleted.
(WebCore::AXAttributeCacheEnabler::AXAttributeCacheEnabler): Deleted.
(WebCore::AXAttributeCacheEnabler::~AXAttributeCacheEnabler): Deleted.
* Source/WebCore/accessibility/AccessibilityMenuList.cpp:
(WebCore::AccessibilityMenuList::didUpdateActiveOption):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::isDetached const):
(WebCore::AccessibilityObject::lineForPosition const):
(WebCore::AccessibilityObject::updateBackingStore):
(WebCore::AccessibilityObject::children):
(WebCore::AccessibilityObject::localizedActionVerb const):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::previousLineStartPosition const):
(WebCore::AccessibilityObject::accessibilityIgnoreAttachment const): Deleted.
(WebCore::AccessibilityObject::accessibilityPlatformIncludesObject const): Deleted.
(WebCore::AccessibilityObject::children): Deleted.
(WebCore::AccessibilityObject::localizedActionVerb const): Deleted.
(WebCore::AccessibilityObject::actionVerb const): Deleted.
(WebCore::AccessibilityObject::lineForPosition const): Deleted.
(WebCore::AccessibilityObject::updateBackingStore): Deleted.
(WebCore::AccessibilityObject::detachPlatformWrapper): Deleted.
* Source/WebCore/accessibility/cocoa/AccessibilityObjectCocoa.mm:
* Source/WebCore/accessibility/ios/AXObjectCacheIOS.mm:
* Source/WebCore/accessibility/ios/AccessibilityObjectIOS.mm:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/AccessibilityObjectMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/accessibility/playstation/AXObjectCachePlayStation.cpp:
* Source/WebCore/accessibility/playstation/AccessibilityObjectPlayStation.cpp:
* Source/WebCore/accessibility/win/AccessibilityObjectWin.cpp:
* Source/WebCore/accessibility/win/AccessibilityObjectWrapperWin.cpp:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::updateTitle):
(WebCore::Document::willBeRemovedFromFrame):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/TextCheckingHelper.cpp:
(WebCore::platformOrClientDrivenTextCheckerEnabled):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::showPopover):
(WebCore::HTMLElement::hidePopoverInternal):
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::HTMLTextFormControlElement::setInnerTextValue):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewAccessible.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebViewAccessible.h:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseDispose):
(webkit_web_view_base_class_init):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::accessible):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::View::~View):
(WKWPE::View::accessible const):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindA11y):
(WebKit::bubblewrapSpawn):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::accessibilityProxy):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::bindAccessibilityTree):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKAccessibilityEnable):
(WKAccessibilityRootObject):
(WKAccessibilityFocusedObject):
(WKAccessibilityFocusedUIElement):
(WKAccessibilityAnnounce):
(WKAccessibilityEnableEnhancedAccessibility):
(WKAccessibilityEnhancedAccessibilityEnabled):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame setAccessibleName:]):
(-[WebFrame enhancedAccessibilityEnabled]):
(-[WebFrame setEnhancedAccessibility:]):
(-[WebFrame accessibilityRoot]):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/DumpRenderTree/AccessibilityController.cpp:
(AccessibilityController::resetToConsistentState):
* Tools/DumpRenderTree/AccessibilityUIElement.cpp:
(AccessibilityUIElement::getJSClass):
* Tools/MiniBrowser/wpe/main.cpp:
(activate):
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.h:
(WTR::AccessibilityController::rootElement): Deleted.
(WTR::AccessibilityController::focusedElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarker.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarkerRange.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.cpp:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityUIElement.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityController.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityTextMarker.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityTextMarkerRange.idl:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/AccessibilityUIElement.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::beginTesting):
(WTR::InjectedBundle::done):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
(WTR::InjectedBundle::textInputController):
(WTR::InjectedBundle::accessibilityController):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::didClearWindowForFrame):
* Tools/WebKitTestRunner/InjectedBundle/mac/AccessibilityUIElementMac.mm:
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityControllerWin.cpp:
* Tools/WebKitTestRunner/InjectedBundle/win/AccessibilityUIElementWin.cpp:
* Tools/wpe/backends/PlatformWPE.cmake:
* Tools/wpe/backends/ViewBackend.cpp:
(WPEToolingBackends::ViewBackend::dispatchInputKeyboardEvent):
(WPEToolingBackends::ViewBackend::initializeAccessibility): Deleted.
(WPEToolingBackends::ViewBackend::updateAccessibilityState): Deleted.
* Tools/wpe/backends/ViewBackend.h:
* Tools/wpe/backends/atk/ViewBackendAtk.cpp:
* Tools/wpe/backends/atk/WebKitAccessibleApplication.cpp:
(webkitAccessibleApplicationSetChild):
* Tools/wpe/backends/atk/WebKitAccessibleApplication.h:

Canonical link: https://commits.webkit.org/273021@main
